Tutorials/GraphHamiltonian/J1J2/j1j2.py

```diff
@@ -0,0 +1,113 @@
+#Copyright 2018 The Simons Foundation, Inc. - All Rights Reserved.
+
+#Licensed under the Apache License, Version 2.0 (the "License");
+#you may not use this file except in compliance with the License.
+#You may obtain a copy of the License at
+
+#    http://www.apache.org/licenses/LICENSE-2.0
+
+#Unless required by applicable law or agreed to in writing, software
+#distributed under the License is distributed on an "AS IS" BASIS,
+#WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#See the License for the specific language governing permissions and
+#limitations under the License.
+
+from __future__ import print_function
+import json
+import numpy as np
+import networkx as nx
+
+#Sigma^z*Sigma^z interactions
+sigmaz = [[1, 0], [0, -1]]
+mszsz = (np.kron(sigmaz, sigmaz))
+
+#Exchange interactions
+exchange = np.asarray([[0, 0, 0, 0], [0, 0, 2, 0], [0, 2, 0, 0], [0, 0, 0, 0]])
+
+#Couplings J1 and J2
+J = [1, 0.4]
+L = 20
+
+pars = {}
+
+# Define bond operators, labels, and couplings
+bond_operator = [
+    (J[0] * mszsz).tolist(),
+    (J[1] * mszsz).tolist(),
+    (-J[0] * exchange).tolist(),
+    (J[1] * exchange).tolist(),
+]
+
+bond_color = [1, 2, 1, 2]
+
+# Define custom graph
+G = nx.Graph()
+for i in range(L):
+    G.add_edge(i, (i + 1) % L, color=1)
+    G.add_edge(i, (i + 2) % L, color=2)
+
+    G.add_edge((i + 1) % L, i, color=1)
```

Comment:
Inserting twice the same edges? I guess the energy should be twice as large then?
I would insert bonds only once, to be consistent with the other hamiltonians

Reply:
Sorry this was left from a test when I initialized the graph as a DiGraph and had to specify each edge in both directions. I will remove it.